src/Application.vala

```diff
@@ -54,6 +56,30 @@ public class Sideload.Application : Gtk.Application {
     }
 
     private async void open_file (File file) {
+        if (file.get_uri ().has_prefix ("flatpak+https://")) {
+            var uri = file.get_uri ().replace ("flatpak+https://", "https://");
+            var path = Path.build_filename (
+                Environment.get_user_special_dir (UserDirectory.DOWNLOAD),
+                Path.get_basename (uri)
+            );
+
+            var remote_file = File.new_for_uri (uri);
+            File local_file = File.new_for_path (path);
+            try {
+                if (!remote_file.copy (local_file, FileCopyFlags.OVERWRITE)) {
+                    warning ("Failed to download file from %s", uri);
+                    release ();
+                    return;
+                }
+            } catch (Error e) {
+                critical ("Failed to download file from %s: %s", uri, e.message);
```

Comment:
I think we probably need to throw an error that the user can see in case of a failure here